### ISSUE TITLE
Report unbound for deferred imported variable

### ISSUE THREAD

**Issue by yanyongyu** [microsoft/pyright#5618]
**Describe the bug**
Pyright 1.1.319 becomes to report the deferred imported variable to be unbound. 1.1.317 is ok.

**Code or Screenshots**

test module imports a variable from an other module later.

```python
def run():
    return a + 1  # error: "a" is unbound

from .test2 import a
```

**VS Code extension or command-line**

command line

-   Pyright version: 1.1.319
-   OS and version: linux x64
-   Python version (and distribution if applicable, e.g. Anaconda): 3.10

**Comment by erictraut** [comment]
Thanks for the bug report. This was introduced as part of the fix for #5548. I'll investigate further.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.320, which I just published. It will also be included in a future version of pylance.
